### ISSUE TITLE
DNS Service crashes on startup

### ISSUE THREAD

**Issue by AyrA** [TechnitiumSoftware/DnsServer#161]
Latest version. Just installed (updated from 4.x)

```
[2020-08-02 22:59:17 UTC] Logging started.
[2020-08-02 22:59:18 UTC] DNS Server config file was loaded: C:\Program Files (x86)\Technitium\DNS Server\config\dns.config
[2020-08-02 22:59:18 UTC] DNS Server successfully loaded zone file: C:\Program Files (x86)\Technitium\DNS Server\config\zones\1.168.192.in-addr.arpa.zone
[2020-08-02 22:59:18 UTC] DNS Server successfully loaded zone file: C:\Program Files (x86)\Technitium\DNS Server\config\zones\archive.is.zone
[2020-08-02 22:59:18 UTC] DNS Server successfully loaded zone file: C:\Program Files (x86)\Technitium\DNS Server\config\zones\archive.today.zone
[2020-08-02 22:59:18 UTC] DNS Server is loading allowed zone file: C:\Program Files (x86)\Technitium\DNS Server\config\allowed.config
[2020-08-02 22:59:18 UTC] DNS Server allowed zone file was loaded: C:\Program Files (x86)\Technitium\DNS Server\config\allowed.config
[2020-08-02 22:59:18 UTC] Failed to start Web Service (v5.0.2.0)
System.IO.IOException: Cannot create a file when that file already exists.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalMove(String sourceFileName, String destFileName, Boolean checkHost)
   at DnsServerCore.Dns.ZoneManagers.BlockedZoneManager.LoadBlockedZoneFile()
   at DnsServerCore.WebService.Start()
[2020-08-02 22:59:18 UTC] System.IO.IOException: Cannot create a file when that file already exists.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalMove(String sourceFileName, String destFileName, Boolean checkHost)
   at DnsServerCore.Dns.ZoneManagers.BlockedZoneManager.LoadBlockedZoneFile()
   at DnsServerCore.WebService.Start()
   at System.ServiceProcess.ServiceBase.ServiceQueuedMainCallback(Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.ServiceProcess.ServiceBase.Run(ServiceBase[] services)
   at DnsService.Program.Main()

```

**Comment by AyrA** [comment]
The logs make it look like the issue is with the blacklists. There's no technical reason for the DNS system itself to fail if the block list handling does. Please make sure that errors in ANYTHING not directly related to operating the Port 53 service will not crash said service.

**Comment by AyrA** [comment]
I think I figured it out. If the file `custom-blocked.config` (1) and `blocked.config` (2) both exist, it crashes. It looks like the service tries to rename File 1 to File 2.

File 1 seems to be empty. Contents: `42 5A 01 00 00 00 00`

**Comment by ShreyasZare** [comment]
Thanks for the feedback. Your analysis is right, the server is renaming the "custom-blocked.config" to "blocked.config" in the new version. I am not really sure how "blocked.config" already exists in your case. Just delete the "blocked.config" and restart the server and that should fix this. I will add the missing error handling for this case in code.

**Comment by AyrA** [comment]
The file exist probably because this is not a fresh install, but an upgrade. The blocked.config is also much larger than the custom-blocked.config so I ended up deleting the custom file instead as it seemed empty.

**Comment by ShreyasZare** [comment]
> 
> 
> The file exist probably because this is not a fresh install, but an upgrade. The blocked.config is also much larger than the custom-blocked.config so I ended up deleting the custom file instead as it seemed empty.

The older version is not supposed to have "blocked.config" in there actually. The new version thus renames "custom-blocked.config".

**Comment by ShreyasZare** [comment]
This bug was fixed in the new released version 5.1.
